.github/workflows/_swift.yml

```diff
@@ -5,172 +5,79 @@ on:
-          INSTALLER_CODE_SIGN_IDENTITY: "3rd Party Mac Developer Installer: Firezone, Inc. (47R2M6779T)"
-        run: |
-          # Use the same Xcode version as development
-          sudo xcode-select -s /Applications/Xcode_${{ matrix.xcode }}.app
-
-          # Copy xcconfig
-          cp Firezone/xcconfig/app_store.xcconfig Firezone/xcconfig/config.xcconfig
-
-          # App Store Connect requires a new build version on each upload and it must be an integer.
-          # See https://developer.apple.com/documentation/xcode/build-settings-reference#Current-Project-Version
-          seconds_since_epoch=$(date +%s)
-          sed -i '' "s/CURRENT_PROJECT_VERSION = [0-9]/CURRENT_PROJECT_VERSION = $seconds_since_epoch/" \
-            Firezone.xcodeproj/project.pbxproj
-
-          # Unfortunately the macOS app requires an installer package to make it into the App Store,
-          # while iOS requires an ipa. The process for building each of these is slightly different.
-          if [ "${{ matrix.platform }}" = "iOS" ]; then
-            # Build archive
-            xcodebuild archive \
-              GIT_SHA=${GITHUB_SHA} \
-              -skipMacroValidation \
-              -archivePath $RUNNER_TEMP/Firezone.xcarchive \
-              -configuration Release \
-              -scheme Firezone \
-              -sdk ${{ matrix.sdk }} \
-              -destination '${{ matrix.destination }}'
-            # Export IPA
-            xcodebuild \
-              -exportArchive \
-              -archivePath $RUNNER_TEMP/Firezone.xcarchive \
-              -exportPath $RUNNER_TEMP/ \
-              -exportOptionsPlist Firezone/ExportOptions.plist
-
-            # Save resulting file to use for upload
-            echo "app_bundle=$RUNNER_TEMP/Firezone.ipa" >> "$GITHUB_OUTPUT"
-          elif [ "${{ matrix.platform }}" = "macOS" ]; then
-            # Build app bundle
-            xcodebuild build \
-              GIT_SHA=${GITHUB_SHA} \
-              -skipMacroValidation \
-              -configuration Release \
-              -scheme Firezone \
-              -sdk ${{ matrix.sdk }} \
-              -destination '${{ matrix.destination }}'
-            # Move it from randomized build output dir to somewhere we can find it
-            mv ~/Library/Developer/Xcode/DerivedData/Firezone-*/Build/Products/Release/Firezone.app $RUNNER_TEMP/.
-            # Create signed installer pkg
-            productbuild \
-              --sign "${{ env.INSTALLER_CODE_SIGN_IDENTITY }}" \
-              --component $RUNNER_TEMP/Firezone.app /Applications $RUNNER_TEMP/Firezone.pkg
-
-            # Save resulting file to use for upload
-            echo "app_bundle=$RUNNER_TEMP/Firezone.pkg" >> "$GITHUB_OUTPUT"
-          else
-            echo "Unsupported platform"
-            exit 1
-          fi
-      - name: Upload build to App Store Connect
+          IOS_APP_PROVISIONING_PROFILE: "${{ secrets.APPLE_IOS_APP_PROVISIONING_PROFILE }}"
```

Comment:
Not every matrix job requires every secret, but doing it like this simplifies the build scripts.

The overall goal here was to not require any ENV vars when building locally, and override appropriate ones from CI if we're building on a runner.